### PR TITLE
Frontend ⟶ Sexpression conversion and testing pass

### PR DESCRIPTION
**Pull request by mariari**
Finishes the Conversion of phase 2 in #671

### CODE REVIEW

File: doc/Architecture/Sexp.org [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+   - output
+     #+begin_src lisp
+       (:defun add (x)
+         (:cond ((:infix == x 3) 1)
+                ((:infix == x 4) 5)
+                (else            0)))
+     #+end_src
+* Notes
+** Design Principles
+Overall the goal of the S-expression incantation is twofold.
```
Comment (cwgoes) [inline doc/Architecture/Sexp.org:307]: incantation? Odd word choice, do you just mean the syntax? Or the system as a whole?

File: doc/Architecture/Sexp.org [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+
+1) An unambiguous syntax that is faithful to the original ML syntax
```
Comment (cwgoes) [inline doc/Architecture/Sexp.org:309]: What does it mean to be faithful? Is there a bijection (ideally)?

**Review by cwgoes — CHANGES_REQUESTED** [review]
Concept ACK, code looks fine, a few questions on the docs

Presumably we should have tests for `transTopLevel` and `transExpr`? Or do you intend to test those in another way or separate PR?

File: doc/Architecture/Sexp.org (hunk 2 above)
Comment (mariari) [reply doc/Architecture/Sexp.org:309]: Yes we preserve properties. Though when I wrote this i was using more colloquial meaning of the word

File: doc/Architecture/Sexp.org (hunk 1 above)
Comment (mariari) [reply doc/Architecture/Sexp.org:307]: As in the expression of this form. Of the reasoning of the form of the syntax, yes.

Part of the reason for the word choice is to pay homage to the imagery given by the dragon book [1] and the legendary SICP course [2]. 

[1] https://en.wikipedia.org/wiki/Compilers:_Principles,_Techniques,_and_Tools
[2] https://www.youtube.com/watch?v=-J_xL4IGhJA&list=PLE18841CABEA24090

**Comment by mariari** [comment]
> Concept ACK, code looks fine, a few questions on the docs
> 
> Presumably we should have tests for `transTopLevel` and `transExpr`? Or do you intend to test those in another way or separate PR?

Yes, as this PR doesn't include the parser, I've put that into a seperate PR... writing tests without the parser is a pain

File: doc/Architecture/Sexp.org (hunk 1 above)
Comment (cwgoes) [reply doc/Architecture/Sexp.org:307]: OK, but we should define it or link to those things, it's not particularly standard usage

File: doc/Architecture/Sexp.org (hunk 2 above)
Comment (cwgoes) [reply doc/Architecture/Sexp.org:309]: Can you be more specific about what properties are preserved?

**Review by cwgoes — APPROVED** [review]
ACK

**Pull request merged**